lib/backend/backend_test.go

```diff
@@ -18,54 +18,49 @@ import (
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 
-	"github.com/projectcalico/libcalico-go/lib/backend/etcd"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	bapi "github.com/projectcalico/libcalico-go/lib/backend/api"
 	"github.com/projectcalico/libcalico-go/lib/backend/model"
 	"github.com/projectcalico/libcalico-go/lib/testutils"
 )
 
-// Setting localhost as the etcd endpoint location since that's where `make run-etcd` runs it.
-var etcdConfig = &etcd.EtcdConfig{
-	EtcdEndpoints: "http://127.0.0.1:2379",
-}
-
-var _ = Describe("Backend tests", func() {
-	etcdClient, _ := etcd.NewEtcdClient(etcdConfig)
-
+var _ = testutils.E2eDatastoreDescribe("Backend tests", testutils.DatastoreEtcdV2, func(config api.CalicoAPIConfig) {
 	var (
-		block model.KVPair
-		err   error
+		block  model.KVPair
+		client bapi.Client
+		err    error
 	)
 
 	BeforeEach(func() {
-		testutils.CleanEtcd()
-
+		c := testutils.CreateCleanClient(config)
+		client = c.Backend
 		block = model.KVPair{
 			Key: model.BlockKey{
 				CIDR: testutils.MustParseNetwork("10.0.0.0/26"),
 			},
-			Value: model.AllocationBlock{
+			Value: &model.AllocationBlock{
 				CIDR: testutils.MustParseNetwork("10.0.0.0/26"),
 			},
 		}
 
-		_, err = etcdClient.Create(&block)
+		_, err = client.Create(&block)
 		Expect(err).NotTo(HaveOccurred())
 	})
 
 	Describe("Create", func() {
 
 		It("persists a new kv pair", func() {
-			kv, err := etcdClient.Get(block.Key)
+			kv, err := client.Get(block.Key)
 			Expect(err).NotTo(HaveOccurred())
 
 			expectedCIDR := kv.Value.(*model.AllocationBlock).CIDR
-			persitedCIRD := block.Value.(model.AllocationBlock).CIDR
+			persitedCIRD := block.Value.(*model.AllocationBlock).CIDR
```

Comment:
I know this is not related to your change.... but would it be possible to rename 

`persitedCIRD -> persistedCIDR`